test/checkpoint.test.js

```diff
@@ -24,5 +25,36 @@ describe('Checkpoint', function() {
         }
       ], done);
     });
+
+    it('Should be no race condition when calling in parallel', function(done) {
+      async.parallel([
+        function(next) { Checkpoint.current(next); },
+        function(next) { Checkpoint.current(next); }
+      ], function(err, list) {
+        if (err) return done(err);
+        Checkpoint.find(function(err, data) {
+          if (err) return done(err);
+          expect(data).to.have.length(1);
+          done();
+        });
+      });
+    });
+
+    it('Checkpoint.current() for non existing checkpoint should initialize checkpoint', function(done) {
+      Checkpoint.current(function(err, seq) {
+        expect(seq).to.equal(1);
+        done(err);
+      });
+    });
+
+    it('bumpLastSeq() works when singleton instance does not exists yet', function(done) {
+      Checkpoint.bumpLastSeq(function(err, cp) {
+        // We expect `seq` to be 2 since `checkpoint` does not exist and
+        // `bumpLastSeq` for the first time not only initializes it to one,
+        // but also increments the initialized value by one.
+        expect(cp.seq).to.equal(2);
```

Comment:
@bajtos, Please note I can check if the `checkpoint` is a new instance or not and if it is new I can just initialize it to 1 as opposed to initialized it and increment the initialized value by one; something like:

```
  Checkpoint.bumpLastSeq = function(cb) {
    var Checkpoint = this;
    Checkpoint._getSingleton(function(err, cp, isNew) {
      if (err) return cb(err);
      if (isNew) return cb(null, cp);
      
      var originalSeq = cp.seq;
      cp.seq++;
      // Update the checkpoint but only if it was not changed under our hands
      Checkpoint.updateAll({id: cp.id, seq: originalSeq}, {seq: cp.seq}, function(err, info) {
        if (err) return cb(err);
        cb(null, cp);
      });
    });
  };
```

But it depends what behavior your would expect in this case...